modules/services/views/src/main/java/au/com/shiftyjelly/pocketcasts/views/multiselect/MultiSelectToolbar.kt

```diff
@@ -30,42 +30,54 @@ class MultiSelectToolbar @JvmOverloads constructor(
 ) : Toolbar(context, attrs, defStyleAttr) {
 
     companion object {
+        private const val MAX_ICONS_BOOKMARKS = 2
         const val MAX_ICONS = 4
     }
 
     private var overflowItems: List<MultiSelectAction> = emptyList()
-    private var fragmentManager: FragmentManager? = null
-    private var multiSelectHelper: MultiSelectHelper? = null
     @Inject lateinit var analyticsTracker: AnalyticsTrackerWrapper
 
-    fun setup(
+    fun <T> setup(
         lifecycleOwner: LifecycleOwner,
-        multiSelectHelper: MultiSelectHelper,
+        multiSelectHelper: MultiSelectHelper<T>,
         @MenuRes menuRes: Int?,
         fragmentManager: FragmentManager
     ) {
-
-        this.fragmentManager = fragmentManager
-        this.multiSelectHelper = multiSelectHelper
-
         setBackgroundColor(context.getThemeColor(UR.attr.support_01))
         if (menuRes != null) {
             inflateMenu(menuRes)
         } else {
             multiSelectHelper.toolbarActions.observe(lifecycleOwner) {
                 menu.clear()
 
-                it.subList(0, MAX_ICONS).forEachIndexed { _, action ->
+                val maxIcons = when (multiSelectHelper) {
+                    is MultiSelectBookmarksHelper -> MAX_ICONS_BOOKMARKS
+                    is MultiSelectEpisodesHelper -> MAX_ICONS
```

Comment:
Just an idea so feel free to ignore it. Would it be better to keep the bookmarks and episodes out of this class. Instead the multiSelectHelper could have an abstract method.
```
val maxIcons = multiSelectHelper.maxIcons
...
abstract val maxIcons: Int
```

Reply:
Nice idea! I applied it in 5a98c7c.